scripts/ci-build.sh

```diff
@@ -376,3 +378,16 @@ fi
 echo Artifacts in staging/
 ls -R staging/
 
+# Restore real user - if this was not defined locally it is either unset again,
+# or defaulting to what the user has defined globally.
+if [ -z "$olduser" ] ; then
+  git config --unset user.name
+else
+  git config user.name "$olduser"
+fi
+if [ -z "$oldemail" ] ; then
+  git config --unset user.email
+else
+  git config user.email "$oldemail"
+fi
+
```

Comment:
A bit over the top perhaps, but I think this kind of "restore" functionality might be nice to add into a function which is run at the end of the script, as well as having some signal handlers setup to restore in case of failure and/or user signalling the script to stop. For example:

```
#!/bin/bash

foo() {
  echo bar
  exit
}

trap "foo" SIGINT SIGTERM

doStuff...

foo
```

Reply:
Updated